workers/generic-worker/insecure.go

```diff
@@ -26,17 +27,41 @@ func secure(configFile string) {
 	log.Printf("WARNING: can't secure generic-worker config file %q", configFile)
 }
 
-func (task *TaskRun) generateInteractiveCommand(ctx context.Context) (*exec.Cmd, error) {
+func (task *TaskRun) generateInteractiveCommand(d2gConversionInfo *d2g.ConversionInfo, ctx context.Context) (*exec.Cmd, error) {
+	var cmd []string
+	var env []string
+
+	if d2gConversionInfo != nil {
+		pathEnv := os.Getenv("PATH")
+		env = []string{"PATH=" + pathEnv}
+
+		cmd = []string{"docker", "exec", "-it", d2gConversionInfo.ContainerName, "/bin/bash"}
+	} else {
+		env = task.EnvVars()
+		cmd = []string{"bash"}
+	}
+
+	return task.newCommandForInteractive(cmd, env, ctx)
+}
+
+func (task *TaskRun) generateInteractiveIsReadyCommand(d2gConversionInfo *d2g.ConversionInfo, ctx context.Context) (*exec.Cmd, error) {
+	pathEnv := os.Getenv("PATH")
+	env := []string{"PATH=" + pathEnv}
+	cmd := []string{"/bin/bash", "-cx", "/bin/[ \"`/usr/bin/docker container inspect -f '{{.State.Running}}' " + d2gConversionInfo.ContainerName + "`\" = \"true\" ]"}
```

Comment:
This is slick!

My only concern here is the path to the `docker` binary. I wonder if some workers may have a different path for some reason. Wdyt about using `which docker` instead of the hardcoded path here?